### PR TITLE
Kamu UI 499 flow config separation

### PR DESCRIPTION
**Pull request by dmitriy-borzenko**
Closes: https://github.com/kamu-data/kamu-web-ui/issues/499

Result:
image

image

image

image

### CODE REVIEW

File: src/app/common/directives/feature-flag.directive.ts [hunk 1]
```diff
@@ -15,6 +15,7 @@ export class FeatureFlagDirective {
 
     @Input("appFeatureFlag") public featureName: string;
 
+    /* istanbul ignore next */
```
Comment (zaychenko-sergei) [inline src/app/common/directives/feature-flag.directive.ts:18]: Why?

File: src/app/common/directives/feature-flag.directive.ts [hunk 2]
```diff
@@ -37,6 +38,7 @@ export class FeatureFlagDirective {
         }
     }
 
+    /* istanbul ignore next */
```
Comment (zaychenko-sergei) [inline src/app/common/directives/feature-flag.directive.ts:41]: Why?

File: src/app/dataset-view/additional-components/dataset-settings-component/services/dataset-scheduling.service.spec.ts [hunk 3]
```diff
@@ -129,24 +153,43 @@ describe("DatasetSchedulingService", () => {
...
+                expect(toastrServiceErrorSpy).toHaveBeenCalledTimes(1);
             });
 
         expect(subscription$.closed).toBeTrue();
-    });
+    }));
+
+    // it("should check set dataset flow batching with error", () => {
+    //     const errorMessage = mockSetDatasetFlowBatchingError.datasets.byId?.flows.configs.setConfigTransform.message;
+    //     spyOn(datasetFlowApi, "setDatasetFlowBatching").and.returnValue(of(mockSetDatasetFlowBatchingError));
```
Comment (zaychenko-sergei) [inline src/app/dataset-view/additional-components/dataset-settings-component/services/dataset-scheduling.service.spec.ts:177]: Garbage

File: src/app/dataset-view/additional-components/dataset-settings-component/tabs/scheduling/dataset-settings-scheduling-tab.component.html [hunk 4]
```diff
@@ -1,167 +1,222 @@
...
+                            type="button"
+                            class="btn btn-success"
+                            (click)="saveIngestConfiguration()"
+                            data-test-id="save-polling-configuration"
+                        >
+                            Save configuration
+                        </button>
+                    </div>
+                </form>
+            </div>
```
Comment (zaychenko-sergei) [inline src/app/dataset-view/additional-components/dataset-settings-component/tabs/scheduling/dataset-settings-scheduling-tab.component.html:144]: I recommend to split this on multiple components. Definitely, trigger choices will be useful in other settings as well, let's say GC schedule in future Admin dashboard, or maybe regular Soft Compacting schedule, once we implement it.

It seems natural to split by main panels: triggers and configuration.

File: src/app/dataset-flow/dataset-flow-details/tabs/flow-details-history-tab/flow-details-history-tab.helpers.ts [hunk 5]
```diff
@@ -64,6 +67,8 @@ export class DatasetFlowDetailsHelpers {
                 return { icon: "downloading", class: "text-muted" };
             case "FlowEventScheduledForActivation":
                 return { icon: "timer", class: "text-muted" };
+            case "FlowConfigSnapshotModified":
+                return { icon: "downloading", class: "text-muted" };
```
Comment (zaychenko-sergei) [inline src/app/dataset-flow/dataset-flow-details/tabs/flow-details-history-tab/flow-details-history-tab.helpers.ts:71]: I think `FlowEventStartConditionUpdated` and `FlowConfigSnapshotModified` should have different icons. Start condition represents state change of the flow while it's waiting (we are waiting for schedule, throttling, executor, ...), and config snapshot represents the changes have been made to the configuration.

File: src/app/common/directives/feature-flag.directive.ts (hunk 1 above)
Comment (dmitriy-borzenko) [reply src/app/common/directives/feature-flag.directive.ts:18]: I thought that this affects the percentage of test coverage, so I deleted it.

File: src/app/common/directives/feature-flag.directive.ts (hunk 2 above)
Comment (dmitriy-borzenko) [reply src/app/common/directives/feature-flag.directive.ts:41]: the same

File: src/app/dataset-flow/dataset-flow-details/tabs/flow-details-history-tab/flow-details-history-tab.helpers.ts (hunk 5 above)
Comment (dmitriy-borzenko) [reply src/app/dataset-flow/dataset-flow-details/tabs/flow-details-history-tab/flow-details-history-tab.helpers.ts:71]: changed icon

File: src/app/dataset-view/additional-components/dataset-settings-component/tabs/scheduling/dataset-settings-scheduling-tab.component.html (hunk 4 above)
Comment (dmitriy-borzenko) [reply src/app/dataset-view/additional-components/dataset-settings-component/tabs/scheduling/dataset-settings-scheduling-tab.component.html]: Fixed

**Pull request merged**